### ISSUE TITLE
NullPointerException when using PinnedAmplitudeClient with enableForegroundTracking()

### ISSUE THREAD

**Issue by dmnugent80** [amplitude/Amplitude-Android#97]
We're getting a NullPointerException when trying to use PinnedAmplitudeClient with a call to enable foreground tracking:

```
PinnedAmplitudeClient.getInstance().enableForegroundTracking(this);
```

I verified that the same crash occurs if you take the tilt maze demo, and switch it to PinnedAmplitudeClient.

stack trace:

```
04-18 17:51:39.859 E/AndroidRuntime( 4012): FATAL EXCEPTION: logThread
04-18 17:51:39.859 E/AndroidRuntime( 4012): Process: com.example.www, PID: 4012
04-18 17:51:39.859 E/AndroidRuntime( 4012): java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Long com.amplitude.api.DatabaseHelper.getLongValue(java.lang.String)' on a null object reference
04-18 17:51:39.859 E/AndroidRuntime( 4012): 	at com.amplitude.api.AmplitudeClient.getLongvalue(AmplitudeClient.java:453)
04-18 17:51:39.859 E/AndroidRuntime( 4012): 	at com.amplitude.api.AmplitudeClient.getLastEventTime(AmplitudeClient.java:466)
04-18 17:51:39.859 E/AndroidRuntime( 4012): 	at com.amplitude.api.AmplitudeClient.isWithinMinTimeBetweenSessions(AmplitudeClient.java:546)
04-18 17:51:39.859 E/AndroidRuntime( 4012): 	at com.amplitude.api.AmplitudeClient.startNewSessionIfNeeded(AmplitudeClient.java:510)
04-18 17:51:39.859 E/AndroidRuntime( 4012): 	at com.amplitude.api.AmplitudeClient$4.run(AmplitudeClient.java:599)
04-18 17:51:39.859 E/AndroidRuntime( 4012): 	at android.os.Handler.handleCallback(Handler.java:739)
04-18 17:51:39.859 E/AndroidRuntime( 4012): 	at android.os.Handler.dispatchMessage(Handler.java:95)
04-18 17:51:39.859 E/AndroidRuntime( 4012): 	at android.os.Looper.loop(Looper.java:148)
04-18 17:51:39.859 E/AndroidRuntime( 4012): 	at android.os.HandlerThread.run(HandlerThread.java:61)
```

Note that the problem only occurs if ```enableForegroundTracking()``` is used.

As a quick fix, it works with an override of the `enableForegroundTracking()` method added to PinnedAmplitudeClient, so that it gives the ```AmplitudeCallbacks``` constructor the PinnedAmplitudeClient instance.

```
public class PinnedAmplitudeClient extends AmplitudeClient {

    public static final String TAG = "com.amplitude.api.PinnedAmplitudeClient";
    private static final AmplitudeLog logger = AmplitudeLog.getLogger();

    //..................................
    
    //This fixes it:
    public AmplitudeClient enableForegroundTracking(Application app) {
        if (isUsingForegroundTracking() || !contextAndApiKeySet("enableForegroundTracking()")) {
            return instance;
        }

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH) {
            Log.d("<><>amplitude", "<><><> amplitude Pinned registerActivityLifecycleCallbacks........");
            app.registerActivityLifecycleCallbacks(new AmplitudeCallbacks(instance));
        }

        return instance;
    }

}
```

**Comment by djih** [comment]
@dmnugent80  thanks for reporting and suggesting a fix! I have reproduced the crash, and verified that the suggested fix works. https://github.com/amplitude/Amplitude-Android/pull/99 will be going into the next release (either tonight or tomorrow)

**Comment by dmnugent80** [comment]
@djih Perfect, thank you!

**Comment by djih** [comment]
This has been patched in v2.7.1, please update thanks!
